crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -5523,6 +5542,19 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
                 .to_class_type(self.db())
                 .is_none_or(|enum_class| !class.is_subclass_of(self.db(), enum_class))
             {
+                let (class, _) = class.class_literal(self.db());
+                if let Some(message) = class.deprecated(self.db()) {
+                    if let Some(builder) = self.context.report_lint(
+                        &crate::types::diagnostic::DEPRECATED,
+                        &*call_expression.func,
+                    ) {
+                        let class_name = class.name(self.db());
+                        let mut diag = builder.into_diagnostic(format_args!(
+                            r#"The class "{class_name}" is deprecated: {message}"#
+                        ));
+                        diag.add_primary_tag(ruff_db::diagnostic::DiagnosticTag::Deprecated);
+                    }
```

Comment:
I don't love this current formatting -- pylance displays

```
The class "{class_name}" is deprecated
    {message}
```

Which is presumably some kind of sub-diagnostic and not literally a newline and indent? But Maybe I'm wrong (tried `diag.info(message)` but that didn't display in vscode).